### PR TITLE
Gm9 romfs and exef

### PR DESCRIPTION
**Pull request by TheGoofBall64**
This pull request adds how to dump romfs and exefs from installed titles and gamecarts. this is to give people a direct guide to link to when someone asks how to dump them for whatever reason (be it making a mod for a game, or something else). It could also be used for dumping other files from titles and gamecarts for whatever reason.

I have seen many people now come into the assistance channels asking how to dump the romfs and exefs. or this just coming up in conversation and asking how to do it

### CODE REVIEW

File: _pages/en_US/godmode9-usage.txt [hunk 1]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
   + **NDS Game Cartridge:** Press (A) on `[TitleID].trim.nds` to select it, then select "Copy to 0:/gm9/out"
 1. Your installable `.cia` or non-installable `.nds` formatted file will be outputted to the `/gm9/out/` folder on your SD card
 
+## Dumping a Game Cartridges romfs/exefs
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:125]: `Game Cartridges` -> `Game Cartridge's`

File: _pages/en_US/godmode9-usage.txt [hunk 2]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+
+1. Insert a game cartrige you wish to dump from
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:127]: `cartrige` -> `cartridge`

File: _pages/en_US/godmode9-usage.txt [hunk 3]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Hover over "GAMECART ()"
+1. Press (A), wait (could take up to 20 seconds)
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:130]: You should instead use the wording used on the rest of this page for gamecart-related stuff: `Navigate to [C:] GAMECART`, and omit `Press (A), wait (could take up to 20 seconds)

File: _pages/en_US/godmode9-usage.txt [hunk 4]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Hover over the bigest file ending with `.3ds` and press (A)
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:131]: "biggest" is misspelled, and you should probably just refer to it as `[TitleID].trim.3ds` instead

File: _pages/en_US/godmode9-usage.txt (hunk 1 above)
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:125]: romfs and exefs aren't exactly things that a normal user would understand without context. Maybe use "dumping content from a game cartridge" or something

File: _pages/en_US/godmode9-usage.txt [hunk 5]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Select "NCCH image options"
+1. Select "Mount image to drive"
+1. Press (A) to continue
+1. Hover over `exefs.bin`, hold (R) and press (A) to open "directory options", select "copy to 0:/gm9/out", do the same with `romfs.bin`
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:135]: Change to "Hover over `exefs.bin`, then hold (R) and press (A) at the same time to open the drive options", and have Directory Options and copying to 0:/gm9/out/ as a separate step.

Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:135]: Honestly a bit unclear what you're trying to do here 'cus all this does is copy the `romfs.bin` and `exefs.bin` files which aren't that useful on their own. The main purpose for this is likely game modding in which case you should unpack the romfs which can be done in GM9

File: _pages/en_US/godmode9-usage.txt [hunk 6]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
 1. Your installable `.cia` formatted file will be outputted to the `/gm9/out/` folder on your SD card
 
+## Dumping a Titles romfs/exefs
+
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Hover over the drive applicable to the type of title you wish to dump files for:
+  + **User Installed `Title: [A:] SYSNAND SD`
+  + **System Title: `[1:] SYSNAND CTRNAND`
+1. Hold (R) and press (A) at the same time to open the drive options
+1. Select “open title manager”
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:157]: capitalize "Open"

File: _pages/en_US/godmode9-usage.txt [hunk 7]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Press (A) to continue
+1. Press (A) on the title you wish to dump the romfs/exefs from 
+  + accent letters will be a `?`, like the e in pokemon
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:160]: "any non-alphanumeric characters" -- Japanese characters also become question marks

File: _pages/en_US/godmode9-usage.txt [hunk 8]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + Titles will have the title ID before the name
+1. Press (A) on "open title folder"
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:162]: capitalize open

File: _pages/en_US/godmode9-usage.txt [hunk 9]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Then press (A) on the bigest file ending in `.app`(the file size is on the right of each file in gm9)
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:163]: Misspelled `biggest`

File: _pages/en_US/godmode9-usage.txt [hunk 10]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + `Byte` is smaller than `kb`, `kb` is smaller than `mb`, `mb` is smaller than `gb`. Example: 1.2 gb is bigger than 1.4 mb and so on
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:164]: I don't think this is necessary

File: _pages/en_US/godmode9-usage.txt (hunk 9 above)
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:163]: Don't need note on where file size is

File: _pages/en_US/godmode9-usage.txt [hunk 11]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
   + **User Installed Title**: `[A:] SYSNAND SD`
   + **System Title**: `[1:] SYSNAND CTRNAND`
 1. Hold (R) and press (A) at the same time to open the drive options
-1. Select "Search for titles"
+1. Select "Open title manager"
 1. Press (A) to continue
 1. Press (A) on the `.tmd` file to select it, then select "TMD file options...", then select "Build CIA (standard)"
 1. Your installable `.cia` formatted file will be outputted to the `/gm9/out/` folder on your SD card
 
+## Dumping a Titles romfs/exefs
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:150]: Any similar comments made for "dumping a cartridge's romfs/exefs" are also relevant here

Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:150]: Titles -> Title's

**Review by lilyuwuu — CHANGES_REQUESTED** [review]

File: _pages/en_US/godmode9-usage.txt (hunk 5 above)
Comment (TheGoofBall64) [reply _pages/en_US/godmode9-usage.txt:135]: gm9 says holding (R) and (A) opens "directory options"

Comment (lilyuwuu) [reply _pages/en_US/godmode9-usage.txt:135]: then word it as "directory options". Just don't have all this in one big step

File: _pages/en_US/godmode9-usage.txt [hunk 12]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Press (A) on the title you wish to dump the romfs/exefs from 
+  + Any non-alphanumeric characters will be a `?`, like the e in pokemon or Japanese characters
+  + Titles will have the title ID before the name
+1. Press (A) on "Open title folder"
+1. Then press (A) on the biggest file ending in `.app`
+1. Select (A) on "NCCH image options"
+1. Select (A) on "Mount image to drive"
+1. Press (A) to continue
+1. Hover over the `exefs` directory, hold (R) and press (A) to open the directory options
+1. select "copy to 0:/gm9/out". Do the same with the `romfs` directory
```
Comment (ItsPizzaTime1501) [inline _pages/en_US/godmode9-usage.txt:168]: "select" should probably be capitalized

File: _pages/en_US/godmode9-usage.txt [hunk 13]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + The dumped files will be in `/gm9/out`
```
Comment (ItsPizzaTime1501) [inline _pages/en_US/godmode9-usage.txt:169]: I'd reword this to "The dumped files will be located in `/gm9/out` on your SD card" but I'm sure lily would have a better suggestion

**Review by ItsPizzaTime1501 — COMMENTED** [review]
Other than a few small things this PR looks pretty good to me.

**Comment by TheGoofBall64** [comment]
yes lily i meant this as help with dumping the stuff for modding games mainly

**Comment by TheGoofBall64** [comment]
now that i think about it, we should either move the dump to CIA guide to a page somewhere on 3ds.hacks.guide. or remove the few cartinstall options on this page. the `.dump` command goes into more detail than this page does for cart installing.

but that's a thing for a different PR if/when its ever done

File: _pages/en_US/godmode9-usage.txt [hunk 14]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + Any non-alphanumeric characters will be a `?`, like the e in pokemon or Japanese characters
+  + Titles will have the title ID before the name
+1. Press (A) on "Open title folder"
+1. Then press (A) on the biggest file ending in `.app`
+1. Select (A) on "NCCH image options"
+1. Select (A) on "Mount image to drive"
+1. Press (A) to continue
+1. Hover over the `exefs` directory, hold (R) and press (A) to open the directory options
+1. Select "copy to 0:/gm9/out". Do the same with the `romfs` directory
+  + The dumped files will be in `/gm9/out` on the sd card
```
Comment (ItsPizzaTime1501) [inline _pages/en_US/godmode9-usage.txt:169]: Capitalize "sd"

File: _pages/en_US/godmode9-usage.txt [hunk 15]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Insert a game cartridge you wish to dump from
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Navigate to `[C:] GAMECART`
+1. Hover over `[TitleID].trim.3ds` and press (A)
+1. Select "NCCH image options"
+1. Select "Mount image to drive"
+1. Press (A) to continue
+1. Hover over the `exefs` directory, hold (R) and press (A) to open the directory options
+1. Select "copy to 0:/gm9/out", do the same with the `romfs` directory
+  + The dumped files will be in `/gm9/out` on the sd card
```
Comment (ItsPizzaTime1501) [inline _pages/en_US/godmode9-usage.txt:136]: Capitalize "sd", also I checked other parts of the guide and they all seem to use "in the /whatever/whatever folder on your SD card" so maybe you could change it to that

**Review by ItsPizzaTime1501 — CHANGES_REQUESTED** [review]
poggu

File: _pages/en_US/godmode9-usage.txt [hunk 16]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+## Dumping a Game Cartridge's contents
+
+1. Insert a game cartridge you wish to dump from
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Navigate to `[C:] GAMECART`
+1. Hover over `[TitleID].trim.3ds` and press (A)
+1. Select "NCCH image options"
+1. Select "Mount image to drive"
+1. Press (A) to continue
+1. Hover over the `exefs` directory, hold (R) and press (A) to open the directory options
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:134]: You need a 'then' between "directory," and "hold"

Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:134]: You should probably either tell people to highlight both exefs and romfs and copy to 0:/gm9/out, or say "Hover over the exefs or romfs directory"

File: _pages/en_US/godmode9-usage.txt [hunk 17]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Select "copy to 0:/gm9/out", do the same with the `romfs` directory
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:135]: Move "do the same with the romfs directory" out of here; this is an odd place for it

File: _pages/en_US/godmode9-usage.txt [hunk 18]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Hover over the drive applicable to the type of title you wish to dump the contents of:
+  + **User Installed `Title: [A:] SYSNAND SD`
+  + **System Title: `[1:] SYSNAND CTRNAND`
+1. Hold (R) and press (A) at the same time to open the drive options
+1. Select “Open title manager”
+1. Press (A) to continue
+1. Press (A) on the title you wish to dump the romfs/exefs from 
+  + Any non-alphanumeric characters will be a `?`, like the e in pokemon or Japanese characters
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:160]: capitalize the P in pokemon

Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:160]: maybe use "will display as a ?" instead of "will be a ?"

File: _pages/en_US/godmode9-usage.txt [hunk 19]
```diff
@@ -105,7 +105,7 @@ This will only work if the Health & Safety injection was performed by GodMode9 (
 1. Press (A) to unlock SysNAND (lvl1) writing, then input the key combo given
 1. Press (A) to relock write permissions if prompted
 
-## Dumping a Game Cartridge
+## Dumping a Game Cartridge as a installable .CIA
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:108]: as a -> as an

File: _pages/en_US/godmode9-usage.txt [hunk 20]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+## Dumping a Game Cartridge's contents
+
+1. Insert a game cartridge you wish to dump from
+1. Press and hold (Start), and while holding (Start), power on your device. This will launch GodMode9
+1. Navigate to `[C:] GAMECART`
+1. Hover over `[TitleID].trim.3ds` and press (A)
+1. Select "NCCH image options"
+1. Select "Mount image to drive"
+1. Press (A) to continue
+1. Highlight the `exefs` and `romfs` directorys by holding (L) and selecting them with (A), then hold (R) and press (A) to open the directory options
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:134]: directorys -> directories

Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:134]: potentially a bit of a mouthful to understand

File: _pages/en_US/godmode9-usage.txt [hunk 21]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+1. Select "copy to 0:/gm9/out"
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:135]: capitalize the C in copy

File: _pages/en_US/godmode9-usage.txt [hunk 22]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + The dumped files will be in the `/gm9/out` folder on your SD card
+
+## Dumping a Title as a installable .CIA
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:138]: as a -> as an

File: _pages/en_US/godmode9-usage.txt [hunk 23]
```diff
@@ -122,18 +122,52 @@ Insert the game cartridge you intend to dump into your device
...
+  + **User Installed `Title: [A:] SYSNAND SD`
+  + **System Title: `[1:] SYSNAND CTRNAND`
+1. Hold (R) and press (A) at the same time to open the drive options
+1. Select “Open title manager”
+1. Press (A) to continue
+1. Press (A) on the title you wish to dump the romfs/exefs from 
+  + Any non-alphanumeric characters will display as a `?`, like the e in Pokemon or Japanese characters
+  + Titles will have the title ID before the name
+1. Press (A) on "Open title folder"
+1. Then press (A) on the biggest file ending in `.app`
```
Comment (lilyuwuu) [inline _pages/en_US/godmode9-usage.txt:163]: omit "Then"

**Review by lilyuwuu — COMMENTED** [review]
No glaring issues that I can find, but I'm not too familiar with specifics about GM9's title manager so I can't verify that the steps related to that are accurate.

**Comment by TheGoofBall64** [comment]
If it helps I actually went through while writing it, then wrote down step by step what I did as I did it. Follow them yourself and you will see

**Comment by TheGoofBall64** [comment]
What's happening with this? Will it go through or be scraped

**Comment by lilyuwuu** [comment]
Closing; reworking contents into dumping-titles-and-game-cartridges.
